### PR TITLE
[Test Framework] Add test scripts for serve.py

### PR DESCRIPTION
**Pull request by Deegue**
Test serve.py with metrics:
image

### CODE REVIEW

**Comment by xwu99** [comment]
@Deegue pls rebase to the latest code

**Comment by Deegue** [comment]
CI passed, gentle ping @xwu99 for review, thanks!

File: .github/workflows/workflow_tests.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ on:
       ci_type:
         type: string
         default: 'pr'
-
```
Comment (xwu99) [inline .github/workflows/workflow_tests.yml:9]: it's OK to leave a blank line there.

File: tests/inference/test_serve.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    if serve_local_only:
+        cmd_serve.append("--serve_local_only")
+    if simple:
+        cmd_serve.append("--simple")
+    if keep_serve_termimal:
+        cmd_serve.append("--keep_serve_termimal")
+
+    result_serve = subprocess.run(cmd_serve, capture_output=True, text=True)
+
+    # TODO: Find a better way to assert the result, like checking processes etc.
```
Comment (xwu99) [inline tests/inference/test_serve.py:138]: It's better to print out the result if we want to check it in the CI before we find a better way to assert.

Comment (Deegue) [reply tests/inference/test_serve.py:121]: added

File: .github/workflows/workflow_tests.yml (hunk 1 above)
Comment (Deegue) [reply .github/workflows/workflow_tests.yml]: reverted

File: tests/inference/test_serve.py (hunk 2 above)
Comment (xwu99) [reply tests/inference/test_serve.py:121]: In addition to check "Error" in the stderr, I think you may also need check the exit code of the launched processes. Successful run is 0. If exception thrown, it should be non-zero. Just may sure we can capture all the errors in the CI. could you try that?

Comment (Deegue) [reply tests/inference/test_serve.py:121]: Sure, returncode assert added and CI passed.

File: tests/inference/test_serve.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    assert "Error" not in result_serve.stderr
+    assert result_serve.returncode == 0
+    print("Asserted no erros in the result log, which is:")
```
Comment (xwu99) [inline tests/inference/test_serve.py:124]:
```suggestion
    print("Output of stderr:")
```

File: tests/inference/test_serve.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    if serve_local_only:
+        cmd_serve.append("--serve_local_only")
+    if simple:
+        cmd_serve.append("--simple")
+    if keep_serve_termimal:
+        cmd_serve.append("--keep_serve_termimal")
+
+    result_serve = subprocess.run(cmd_serve, capture_output=True, text=True)
+
+    # TODO: Find a better way to assert the result, like checking processes etc.
```
Comment (xwu99) [inline tests/inference/test_serve.py:121]:
```suggestion
```

**Review by xwu99 — APPROVED** [review]

**Comment by Deegue** [comment]
@xwu99 Thanks for your review! Auto merge when CI is passed.

**Pull request merged**